iothub/device/src/IotHubBaseClient.cs

```diff
@@ -194,7 +199,12 @@ public async Task SendTelemetryBatchAsync(IEnumerable<TelemetryMessage> messages
         /// Calling this API more than once will result in the callback set last overwriting any previously set callback.
         /// A method callback can be unset by setting <paramref name="messageCallback"/> to null.
         /// </remarks>
-        /// <param name="messageCallback">The callback to be invoked when a cloud-to-device message is received by the client.</param>
+        /// <param name="messageCallback">
+        /// The callback to be invoked when a cloud-to-device message is received by the client.
+        /// This user-supplied callback is awaited by the SDK. All of requests will be processed as they arrive.
+        /// Exceptions will be thrown to the users and the SDK won't log them internally, so the user is responsible
```

Comment:
What does this mean: "Exceptions will be thrown to the users"?

Reply:
we don't catch any exceptions in this method of subscribing incoming message callback `SetIncomingMessageCallbackAsync()`

Reply:
we don't catch or handle any exceptions in the method `OnMessageReceivedAsync()`

Reply:
Okay, the original text sounds like we'll throw an exception that the user must catch. Is that the case? Because, if this is a callback, there's no way they can catch it. Exceptions won't be thrown **to** them, right?

Reply:
Also, thanks for documenting, but can we change the behavior to catch exceptions and then update the docs to indicate that? I'd rather fix the behavior than just document bad behavior.

Reply:
Yeah, sure, we should be able to apply the same try-catch logic from `OnMethodCalledAsync()` into this one.